lib/components/component.dart

```diff
@@ -10,12 +11,18 @@ import '../extensions/vector2.dart';
 /// Anything that either renders or updates can be added to the list on [BaseGame]. It will deal with calling those methods for you.
 /// Components also have other methods that can help you out if you want to overwrite them.
 abstract class Component {
+  /// The effects that should run on the component
+  final List<ComponentEffect> _effects = [];
+
   /// This method is called periodically by the game engine to request that your component updates itself.
   ///
   /// The time [t] in seconds (with microseconds precision provided by Flutter) since the last update cycle.
   /// This time can vary according to hardware capacity, so make sure to update your state considering this.
   /// All components on [BaseGame] are always updated by the same amount. The time each one takes to update adds up to the next update cycle.
-  void update(double t);
+  void update(double dt) {
```

Comment:
if we are doing this, we must add @mustCallSuper here

what are the use cases? are we planning to add any non-position effects?

Reply:
Exactly, an example could be fading colors or a user could do pretty much anything with it's own implemented component.
Indeed it should have @mustCallSuper there, good catch!